homeassistant/components/weatherflow_cloud/__init__.py

```diff
@@ -2,30 +2,127 @@
+    WeatherFlowCloudUpdateCoordinatorREST,
+)
 
 PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.WEATHER]
 
 
+@dataclass
+class WeatherFlowCoordinators:
+    """Data Class for Entry Data."""
+
+    rest: WeatherFlowCloudUpdateCoordinatorREST
+    wind: WeatherFlowCloudDataCallbackCoordinator[
+        EventDataRapidWind, RapidWindListenStartMessage, RapidWindWS
+    ]
+    observation: WeatherFlowCloudDataCallbackCoordinator[
+        WebsocketObservation, ListenStartMessage, WebsocketObservation
+    ]
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up WeatherFlowCloud from a config entry."""
 
-    data_coordinator = WeatherFlowCloudDataUpdateCoordinator(hass, entry)
-    await data_coordinator.async_config_entry_first_refresh()
+    LOGGER.debug("Initializing WeatherFlowCloudDataUpdateCoordinatorREST coordinator")
+
+    rest_api = WeatherFlowRestAPI(
+        api_token=entry.data[CONF_API_TOKEN], session=async_get_clientsession(hass)
+    )
+
+    stations = await rest_api.async_get_stations()
+
+    # Define Rest Coordinator
+    rest_data_coordinator = WeatherFlowCloudUpdateCoordinatorREST(
+        hass=hass, config_entry=entry, rest_api=rest_api, stations=stations
+    )
+
+    # Initialize the stations
+    await rest_data_coordinator.async_config_entry_first_refresh()
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = data_coordinator
+    # Construct Websocket Coordinators
+    LOGGER.debug(
+        "Initializing WeatherFlowCloudDataUpdateCoordinatorWebsocketWind coordinator"
+    )
+    websocket_device_ids = rest_data_coordinator.device_ids
+
+    # Build API once
+    websocket_api = WeatherFlowWebsocketAPI(
+        access_token=entry.data[CONF_API_TOKEN], device_ids=websocket_device_ids
+    )
+
+    websocket_observation_coordinator = WeatherFlowCloudDataCallbackCoordinator[
+        WebsocketObservation, WebsocketObservation, ListenStartMessage
+    ](
+        hass=hass,
+        config_entry=entry,
+        rest_api=rest_api,
+        websocket_api=websocket_api,
+        stations=stations,
+        listen_request_type=ListenStartMessage,
+        event_type=EventType.OBSERVATION,
+    )
+
+    websocket_wind_coordinator = WeatherFlowCloudDataCallbackCoordinator[
+        EventDataRapidWind, EventDataRapidWind, RapidWindListenStartMessage
+    ](
+        hass=hass,
+        config_entry=entry,
+        stations=stations,
+        rest_api=rest_api,
+        websocket_api=websocket_api,
+        listen_request_type=RapidWindListenStartMessage,
+        event_type=EventType.RAPID_WIND,
+    )
+
+    # Run setup method.
+    await asyncio.gather(
+        websocket_wind_coordinator.async_setup(),
+        websocket_observation_coordinator.async_setup(),
+    )
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = WeatherFlowCoordinators(
+        rest_data_coordinator,
+        websocket_wind_coordinator,
+        websocket_observation_coordinator,
+    )
```

Comment:
In a follow-up PR, I think `weatherflow_cloud` could be migrated to store the `WeatherFlowCoordinators` object as config entry runtime data (maybe this has already been discussed).